toc/rfc/rfc-draft-ssh-over-ws.md

```diff
@@ -0,0 +1,329 @@
+# Meta
+[meta]: #meta
+- Name: CF SSH over WebSockets
+- Start Date: 2024-04-08
+- Author(s): @domdom82
+- Status: Draft
+- RFC Pull Request: https://github.com/cloudfoundry/community/pull/807
+
+
+## Summary
+
+This RFC aims at providing an additional way of using CF SSH in an effort to reduce the potential attack surface against SSH.
+
+By tunneling SSH through a WebSocket connection, the SSH client may reuse the same port as regular HTTP traffic, removing the need for port 2222.
+
+## Problem
+
+Cloud Foundry is a shared multi-tenant platform.
+
+If a feature requiring an open port such as CF SSH is used by at least one customer, the port has to be opened on the whole platform. It cannot be enabled for one tenant but disabled for another.
+
+In production deployments of Cloud Foundry, it is commonplace for customers to run security scans on their workloads to ensure compliance and reduce attack surfaces.
+
+These scans usually complain that besides port 443 for HTTP traffic, also port 2222 is open for SSH on the internet-facing load balancer. Many customers demand closure of the port as they
+view it as a potential attack surface for brute-force attacks against the SSH protocol.
+
+The issue is exacerbated by features like "bring-your-own-domain" where customers provide a CNAME entry alongside a matching x509 certificate that points their own (sub-)domain to an app hosted on the shared Cloud Foundry environment.
+
+This can cause potential reputation damage to the customer because to an outsider, it looks like the port 2222 is open on the customer's domain, e.g. `www.big-corp.com:2222` open looks much worse than `my-app.cfapps.eu10.hana.ondemand.com:2222` even though both are technically the same.
+
```

Comment:
Ssh ingress has been hard-coded to be handled by a different host: https://github.com/cloudfoundry/capi-release/blob/main/jobs/cloud_controller_ng/templates/cloud_controller_ng.yml.erb#L126

Typically handled by a different load balancer. So I don't understand how custom domains are affected.

Reply:
That may be a SAP-thing. We usually use one LB for all of the platform ingress and let customers point their DNS to that LB, so implicitly all open ports for that LB will show up on the custom domain. It could be done differently, true, I will change the section to reflect that.

Reply:
I've added a section to reflect different options for ingresses. Both have advantages and costs. I've also removed SAP-specific references.